File: Versionen/Datenstrukturen (linear)/BlueJ/04-Stack_ML/Stack.java

public class Stack<ContentType> {

    private StackNode<ContentType> head = null;

    public Stack() {
    }


    public void push( ContentType pContentObject ) {
        StackNode<ContentType> n = new StackNode<ContentType>(pContentObject);
        if( head == null ) {
            head = n;
        } else {
            n.setNext(head);
            head = n;
        }
    }

    public void pop() {
        if( head != null ) {
            head = head.getNext();
        }
    }

    /**
     * Liefert das Inhaltsobjekt vom obersten Knoten des Stapels
     * vom Typ ContentType, falls die Schlange nicht leer ist,
     * ansonsten null.
     *
     * @return Das Inhaltsobjekt oder null.
     */
    public ContentType top() {
        if( !isEmpty() ) {
            return head.getContent();
        } else {
            return null;
        }
    }

    /**
     * Prüft, ob der Stapel leer ist.
     *
     * @return true, wenn der Stapel keine Knoten enthaelt
     */
    public boolean isEmpty() {
        return (head == null);
    }

}

File: Abiturklassen/Python/stack.py
# -*- coding: utf-8 -*-


class Stack(object):
    class StackNode(object):
        def __init__(self, pContent):
            self._content = pContent
            self._nextNode = None

        def setNext(self, pNext):
            self._nextNode = pNext

        def getNext(self):
            return self._nextNode

        def getContent(self):
            return self._content

    def __init__(self):
        self._head = None

    def isEmpty(self):
        return self._head is None

    def push(self, pContent):
        if pContent is not None:
            node = self.StackNode(pContent)
            node.setNext(self._head)
            self._head = node

    def pop(self):
        if not self.isEmpty():
            self._head = self._head.getNext()

    def top(self):
        if not self.isEmpty():
            return self._head.getContent()
        else:
            return None
